### ISSUE TITLE
openssl was built without bitcode. You must rebuild it with bitcode enabled 

### ISSUE THREAD

**Issue by AdrianaPineda** [FredericJacobs/OpenSSL-Pod#13]
I'm getting this warning: 
URGENT: all bitcode will be dropped because [path to openssl class] was built without bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. Note: This will be an error in the future.

Can you please update the version to one that contains bitcode?

**Comment by krzyzanowskim** [comment]
This is Xcode 7 beta feature

**Comment by AdrianaPineda** [comment]
Yeah, its for Xcode 7 beta

**Comment by massar** [comment]
Apple is moving to LibreSSL... likely APIs will become incompatible at one point too....

**Comment by FredericJacobs** [comment]
@massar I don't think Apple wants dynamic linking on iOS.

Still researching the implications of Bitcode. I read about optimizations of intermediate representation code, that can be potentially an issue for time-safe functions.

**Comment by honkmaster** [comment]
Any updates on this issue?

**Comment by honkmaster** [comment]
Maybe this site is helpful: https://forums.developer.apple.com/thread/8810 (Apple developer forum). It is necessary to add '-fembed-bitcode' to the compiler flags.

**Comment by chrisballinger** [comment]
@honkmaster Thank you for the tip!

**Comment by AdrianaPineda** [comment]
@honkmaster thanks I'll check it out

**Comment by honkmaster** [comment]
Here is a snippet of @steipete with a working script at the end: https://gist.github.com/steipete/99f8aa8d6527fa027fd6

**Comment by AdrianaPineda** [comment]
Hi! any update on this? -fembed-bitcode didnt work for me :S

**Comment by FredericJacobs** [comment]
I think it's unwise to compile OpenSSL with Bitcode. Not knowing what compiler optimizations will be performed makes me really scared. Not planning to support this on this pod since it would help developers to shoot themselves in the foot.

**Comment by AdrianaPineda** [comment]
The problem is that for watchOS apps, bitcode is mandatory. I'm stuck here, this means I can't submit my app :S

**Comment by Jeff-F** [comment]
The problem is that for watchOS apps, bitcode is mandatory. I'm stuck here, this means I can't submit my app :S   --  Me too!

**Comment by steipete** [comment]
Simply use the script/gist from my account that was linked here already. The world doesn't end at Cocoapods, just manually build the files for a quick solution until this is resolved.

**Comment by FredericJacobs** [comment]
Here are some of the reasons why I don't think 

(Granted, some arguments are less applicable than others. But introduction of side-channels is a critical issue for a crypto implementation)

**Comment by tyridw** [comment]
This is a late reply, so I suppose a lot has changed since 2015. But I ran into the same problem as initially stated, but I resolved the "You must rebuild it with bitcode enabled" problem by

1. Removing 'OpenSSL' and running pod deintegrate, pod clean etc., see
https://stackoverflow.com/questions/13751147/remove-or-uninstall-library-previously-added-cocoapods

2. Instead installing 'OpenSSL-Universal'
https://cocoapods.org/pods/OpenSSL-Universal

Using Xcode version 10.0

(Disclaimer: I'm not involved in this space. I don't mean to step on anyone's toes if there are contentious issues between the  projects OpenSSL and OpenSSL-Universal)
